src/libraries/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -652,6 +653,10 @@ public sealed partial class Utf8JsonWriter : System.IAsyncDisposable, System.IDi
         public void WriteStringValue(System.ReadOnlySpan<char> value) { }
         public void WriteStringValue(string? value) { }
         public void WriteStringValue(System.Text.Json.JsonEncodedText value) { }
+        public void WriteStringValueSegment(System.ReadOnlySpan<byte> utf8Value, bool isFinalSegment) { }
+        public void WriteStringValueSegment(ReadOnlySpan<char> value, bool isFinalSegment) { }
+        public void WriteStringValueSegment(string? value, bool isFinalSegment) { }
+        public void WriteStringValueSegment(System.Text.Json.JsonEncodedText value, bool isFinalSegment) { }
```

Comment:
Fair - I figured I'd align with the existing interfaces but that is indeed creep on the approved shape. I'll make sure this conforms.

Reply:
`WriteBase64StringValueSegment` is quite a bit more complex than the others as well, because the encoding of segment 1 changes depending on segment 2. Would it be acceptable to scope this PR down to implementing the string variants and make base64 a follow up?

Reply:
Sure.